src/script.js

```diff
@@ -0,0 +1,162 @@
+(function() {
+var mapDraw = function(layers) {
+
+  layers = layers || [];
+
+  if (slideshow.current > -1) {
+    let nextlayers = Object.keys(layers);
+  } else {nextlayers = []}
+
+  // remove existing map layers
+  map.eachLayer(function(leaf) {
+    if (leaf != Stamen_TonerLite) {
+      map.removeLayer(leaf);
+    }
+  });
+
+  // add new layers
+  let names = Object.keys(layers); // layers is list of layers in each slide
+  slideshow.currentlayers = names;
+
+  names.forEach(function(name) {
+    let layer = layers[name];
+    if (!layer) {return;}
+
+    slideshow.leaves.forEach(function(leave){ // leave.leaf is actual leaflet object
+      if (name !== leave.name) {return;}
+
+      leave.leaf.eachLayer(function(l) {
+        if (layer.popup) {
+          l.bindPopup(eval(layer.popup));
```

Comment:
Is code injection like this this considered bad practice? (See lines 43-51 in [`slides.js`](https://github.com/MUSA611-CPLN692-spring2019/cpln692-Midterm/pull/11/files#diff-69db8d1fcaf391379fce03d782432fef) for an example of code to be injected.)

Reply:
In general, `eval` is seen as a dirty trick and a tool of last resort. Judging from the context, I think you could pull off the same behavior by parameterizing your `layer.popup` function on the `l` that you're calling `bindPopup` on rather than evaluating the code.

In general, you're only buying yourself delayed evaluation by using `eval`. This turns out to not really buy you much that a function can't. "Don't read/interpret this code until you're in the context I want" is the rough equivalent of "don't run the code inside this function without injecting the context through a parameter"